backend/prompt_studio/prompt_studio_index_manager/models.py

```diff
@@ -90,31 +91,40 @@ class Meta:
         ]
 
 
+def delete_from_vector_db(index_ids_history, vector_db_instance_id):
+    org_schema = connection.tenant.schema_name
+    util = PromptIdeBaseTool(log_level=LogLevel.INFO, org_id=org_schema)
+    vector_db = VectorDB(
+        tool=util,
+        adapter_instance_id=vector_db_instance_id,
+    )
+    for index_id in index_ids_history:
+        logger.debug(f"Deleting from VectorDB - index id: {index_id}")
+        try:
+            vector_db.delete(ref_doc_id=index_id)
+        except Exception as e:
+            # Log error and continue with the next index id
+            logger.error(f"Error deleting index: {index_id} - {e}")
+
+
 # Function will be executed every time an instance of IndexManager is deleted.
-@receiver(post_delete, sender=IndexManager)
+@receiver(pre_delete, sender=IndexManager)
```

Comment:
@athul-rs how will a `pre_delete` help our case? I believe these signals are async operations right anyway?

Reply:
I have mentioned in the `how` part above, adding the same here:
- The earlier failure was happening in the case of the entire tool delete because during the cascade delete from custom_tool it was clearing all the related before reaching IndexManager thus missing the related ProfileManager entry for matching vector_store_id.